src/charm.py

```diff
@@ -2379,8 +2380,9 @@ def relations_user_databases_map(self) -> dict:
                 logger.debug(f"User {user} has no databases to connect to")
 
         # Copy relations users directly instead of waiting for them to be created
+        custom_username_mapping = self.postgresql_client_relation.get_username_mapping()
```

Comment:
Can't get the custom username from the followers here, so using a cache field in the app secret.